packages/core/src/RenderingEngine/Viewport.ts

```diff
@@ -1259,6 +1261,19 @@ class Viewport implements IViewport {
     };
   }
 
+  /**
+   * Get the camera's current state
+   * @returns The camera object.
+   */
+  public getCamera(): ICamera {
+    const camera = this.getCameraNoRotation();
+
+    return {
```

Comment:
You can directly modify the return from getCameraNoRotation as long as you specify that it needs to always return a new instance - given how much this gets called, that is a reasonable optimization.